botcraft/src/AI/Tasks/InventoryTasks.cpp

```diff
@@ -325,8 +325,51 @@ namespace Botcraft
             Position(1, 0, 0), Position(-1, 0, 0)
             });
 
-        // Check if block is air
         bool midair_placing = true;
+        // If no face specified
+        if (!face.has_value())
+        {
+            if (allow_midair_placing) // Then we don't care and default to Up
+            {
+                face = PlayerDiggingFace::Up;
+            }
+            else
+            {
+                std::vector<PlayerDiggingFace> face_candidates; // Faces next to a block
+                face_candidates.reserve(6);
+                for (int face_idx = 0; face_idx < 6; face_idx++)
+                {
+                    std::lock_guard<std::mutex> world_guard(world->GetMutex());
+                    const Block* neighbour_block = world->GetBlock(pos + neighbour_offsets[face_idx]);
+                    if (neighbour_block && !neighbour_block->GetBlockstate()->IsAir())
+                    {
+                        face_candidates.push_back(static_cast<PlayerDiggingFace>(face_idx));
+                    }
+                }
+                if (face_candidates.size() == 0)
+                {
+                    LOG_WARNING("Can't place a block in midair at " << pos);
+                    return Status::Failure;
+                }
+                Vector3<double> player_orientation;
+                {
+                    std::lock_guard<std::mutex> lock(local_player->GetMutex());
+                    player_orientation = local_player->GetFrontVector();
+                }
+                std::sort( // Find the face face closest to player looking direction
+                    face_candidates.begin(), face_candidates.end(), [&](PlayerDiggingFace a, PlayerDiggingFace b) -> bool
+                    {
+                        Vector3<double> a_offset = neighbour_offsets[static_cast<int>(a)];
+                        Vector3<double> b_offset = neighbour_offsets[static_cast<int>(b)];
+                        return player_orientation.dot(a_offset) > player_orientation.dot(b_offset); 
```

Comment:
Ok that's seems good knowing the fact that the player is always looking at the block center. However, your previous commit had a < b and now a > b. Are you sure this one is the right one based on actual tests? If you tested it that should be good to be merged!